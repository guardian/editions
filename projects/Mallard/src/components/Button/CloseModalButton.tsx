import React from 'react'
import { StyleSheet } from 'react-native'
import { Button } from './Button'
import { color } from 'src/theme/color'

const closeModalButtonStyles = (bgColor?: string, borderColor?: string) =>
    StyleSheet.create({
        dismissText: {
            color: bgColor ? 'white' : color.primary,
        },
        dismissButton: {
            paddingHorizontal: 0,
            backgroundColor: bgColor ? bgColor : 'transparent',
            borderColor: borderColor ? borderColor : color.primary,
            borderWidth: 1,
        },
    })

const CloseModalButton = ({
    onPress,
    bgColor,
    borderColor,
}: {
    onPress: () => void
    bgColor?: string
    borderColor?: string
}) => {
    return (
        <Button
            icon=""
            alt="Dismiss"
            buttonStyles={[
                closeModalButtonStyles(bgColor, borderColor).dismissButton,
            ]}
            textStyles={
                closeModalButtonStyles(bgColor, borderColor).dismissText
            }
            onPress={onPress}
        />
    )
}

export { CloseModalButton }
