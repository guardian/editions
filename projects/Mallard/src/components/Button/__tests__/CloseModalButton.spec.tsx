import React from 'react'
import TestRenderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { CloseModalButton } from '../CloseModalButton'

describe('CloseModalButton', () => {
    it('should display a CloseModalButton', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <CloseModalButton onPress={() => {}} />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
    it('should display a CloseModalButton with a colour prop', () => {
        const component: ReactTestRendererJSON | null = TestRenderer.create(
            <CloseModalButton
                onPress={() => {}}
                bgColor="#41A9E0"
                borderColor="#41A9E0"
            />,
        ).toJSON()
        expect(component).toMatchSnapshot()
    })
})
