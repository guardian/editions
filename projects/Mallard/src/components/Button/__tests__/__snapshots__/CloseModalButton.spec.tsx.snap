// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseModalButton should display a CloseModalButton 1`] = `
<View
  accessibilityHint="Dismiss"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 999,
          "height": 42,
          "justifyContent": "center",
          "paddingHorizontal": 21,
        },
        Object {
          "backgroundColor": "#ffe500",
        },
        Object {
          "alignItems": "center",
          "aspectRatio": 1,
          "justifyContent": "center",
          "paddingHorizontal": 0,
          "width": 42,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderColor": "#052962",
            "borderWidth": 1,
            "paddingHorizontal": 0,
          },
        ],
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "GuardianIcons-Regular",
            "fontSize": 20,
            "lineHeight": 20,
          },
          Array [
            Object {
              "color": "#121212",
            },
            Object {
              "color": "#052962",
            },
          ],
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`CloseModalButton should display a CloseModalButton with a colour prop 1`] = `
<View
  accessibilityHint="Dismiss"
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  hitSlop={
    Object {
      "bottom": 10,
      "left": 10,
      "right": 10,
      "top": 10,
    }
  }
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 999,
          "height": 42,
          "justifyContent": "center",
          "paddingHorizontal": 21,
        },
        Object {
          "backgroundColor": "#ffe500",
        },
        Object {
          "alignItems": "center",
          "aspectRatio": 1,
          "justifyContent": "center",
          "paddingHorizontal": 0,
          "width": 42,
        },
        Array [
          Object {
            "backgroundColor": "#41A9E0",
            "borderColor": "#41A9E0",
            "borderWidth": 1,
            "paddingHorizontal": 0,
          },
        ],
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "fontFamily": "GuardianIcons-Regular",
            "fontSize": 20,
            "lineHeight": 20,
          },
          Array [
            Object {
              "color": "#121212",
            },
            Object {
              "color": "white",
            },
          ],
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
